src/views/ModelPlan/TaskList/ITSolutions/Subtasks/index.tsx

```diff
@@ -0,0 +1,566 @@
+                  data-testid="success-subtask-alert"
+                  className="margin-y-4"
+                >
+                  {t('successMessage')}
+                </Alert>
+              );
+              history.push(
+                `/models/${modelID}/task-list/it-solutions/${operationalNeedID}/${operationalSolutionID}/solution-details`
+              );
+            }
+          })
+          .catch(errors => {
+            formikRef?.current?.setErrors(errors);
+          })
+      );
+    } else {
+      subtasks.map(subtask =>
+        update({
+          variables: {
+            inputs: [
+              {
+                id: subtask.id,
+                changes: {
+                  name: subtask.name,
+                  status: subtask.status
+                }
+              }
+            ]
+          }
+        })
+          .then(response => {
+            if (!response?.errors) {
+              showMessageOnNextPage(
+                <Alert
+                  type="success"
+                  slim
+                  data-testid="success-subtask-alert"
+                  className="margin-y-4"
+                >
+                  {t('successfulUpdateMessage')}
+                </Alert>
+              );
+              history.push(
+                `/models/${modelID}/task-list/it-solutions/${operationalNeedID}/${operationalSolutionID}/solution-details`
+              );
+            }
+          })
+          .catch(errors => {
+            formikRef?.current?.setErrors(errors);
+          })
+      );
+    }
+  };
+
+  const renderModal = () => {
+    return (
+      <Modal isOpen={isModalOpen} closeModal={() => setModalOpen(false)}>
+        <PageHeading headingLevel="h2" className="margin-y-0">
+          {t('removeModal.header', {
+            subTaskName: inputName
+          })}
+        </PageHeading>
+        <p className="margin-top-2 margin-bottom-3">
+          {t('removeModal.warning')}
+        </p>
+        <Button
+          type="button"
+          className="margin-right-4"
+          onClick={() => {
+            handleDelete(inputName, inputId);
+            setModalOpen(false);
+          }}
+        >
+          {t('removeModal.removeSubtask')}
+        </Button>
+        <Button type="button" unstyled onClick={() => setModalOpen(false)}>
+          {t('removeModal.keepSubtask')}
+        </Button>
+      </Modal>
+    );
+  };
+
+  const breadcrumbs = [
+    { text: h('home'), url: '/' },
+    { text: h('tasklistBreadcrumb'), url: `/models/${modelID}/task-list/` },
+    {
+      text: t('itSolutionsTrackerBreadcrumb'),
+      url: `/models/${modelID}/task-list/it-solutions`
+    },
+    {
+      text: t('solutionDetails'),
+      url: `/models/${modelID}/task-list/it-solutions/${operationalNeedID}/${operationalSolutionID}/solution-details`
+    },
+    { text: managingSubtasks ? t('manageSubtasks') : t('addSubtask') }
+  ];
+
+  if (error || !solution) {
+    return <NotFound />;
+  }
```

Comment:
This may be faulty, as `solution` could still be in process of being fetched, and `NotFound` would render.  Checking to see if loading is false in addition ^  would be more accurate.

Reply:
Or have them separated out into error render component, and loading render component